delete-serverless-service-build-pack/Jenkinsfile

```diff
@@ -15,108 +15,74 @@ import groovy.transform.Field
 @Field def configModule
 @Field def configLoader
 @Field def scmModule
+@Field def events
+@Field def serviceMetadataLoader
 
-@Field def Util
 @Field def serviceMetadata
 @Field def g_base_url = ''
 @Field def g_svc_admin_cred_ID = 'SVC_ADMIN'
 @Field def auth_token = ''
-@Field def domain = ''
-@Field def service = ''
-@Field def owner = ''
-@Field def var_db_service_id = ''
-@Field def role = ''
+@Field def service_config
 
 node {
 	echo "Starting delete service job with params : $params"
 
 	jazzBuildModuleURL = getBuildModuleUrl()
-	loadConfigModule(jazzBuildModuleURL)
-	loadSCMModule(jazzBuildModuleURL)
+	loadBuildModule(jazzBuildModuleURL)
 
 	g_base_url = "https://${configLoader.AWS.API.PROD_ID}.execute-api.${configLoader.AWS.REGION}.amazonaws.com/prod"
-	role = configLoader.AWS.ROLEID
 	
-	def service_name
 	def version
 	def repo_name
 	def tracking_id
-	def config
-	def runtime
-	def serviceConfig
-	def domainConfig
 	def cloudfrontEnabled
-	def flowType
-	
-	var_db_service_id = params.db_service_id
+	def flowType	
+	def service_id = params.db_service_id
 
-	if (params.service_name) {
-		service_name =  params.service_name.trim()
-		service = service_name
-	}
-	if(params.domain){
-		domain = params.domain.trim()
-	} else { domain = "" }
 	if(params.version){
 		version = params.version.trim()
 	}
+
 	if(params.tracking_id){
 		tracking_id = params.tracking_id.trim()
-	}
-	repo_name = service_name
-	if(domain && domain != ""){
-		repo_name = domain + "_" + service_name
-	}
+	}	
 
 	auth_token = setCredentials()
-	loadServiceInfo()
-
-	sh 'rm -rf ' + repo_name
-	sh 'mkdir ' + repo_name
 
 	stage("Initialization") {
 
-		checkoutSCM(repo_name)
-		def configObj = dir(repo_name){
-			return LoadConfiguration()
-		}
-		
-		if(configObj.service_id){	
-			def url = "$g_base_url/jazz/services/"+configObj.service_id
-			serviceMetadata = loadServiceMetadataModule(jazzBuildModuleURL)
-			serviceMetadata.setUrl(url)
-			def serviceData = serviceMetadata.loadServiceMetaData()
-			if(serviceData && serviceData.data){	
-				config = serviceData.data['metadata']
-			}else{
-				error "Error while fetching service metadata"
-			}
+		if(service_id){	
+			service_config = serviceMetadataLoader.loadServiceMetadata(service_id)
 		}else{
 			error "Service Id is not available."
 		}
 	
-		runtime = config['providerRuntime']
-		serviceConfig = config['service']
-		domainConfig = config['domain']
-		updateServiceInDB(var_db_service_id, configLoader.AWS.API.PROD_ID, configLoader.AWS.REGION, "deletion_started");
+		if(!events) { error "Can't load events module"	} //Fail here
+		events.setUrl(g_base_url + "jazz/events")
```

Comment:
create initialize method in events module and set all the params as part of one call.